libraries/chain/webassembly/runtimes/eos-vm.cpp

```diff
@@ -513,10 +513,10 @@ REGISTER_HOST_FUNCTION(get_permission_last_used);
 REGISTER_HOST_FUNCTION(get_account_creation_time);
 
 // authorization api
-REGISTER_HOST_FUNCTION(require_auth);
-REGISTER_HOST_FUNCTION(require_auth2);
+REGISTER_HOST_FUNCTION(require_auth, action_check);
+REGISTER_HOST_FUNCTION(require_auth2, action_check);
 REGISTER_HOST_FUNCTION(has_auth);
```

Comment:
If `require_auth` not allowed, why is `has_auth`? Isn't `require_auth` equivalent of `check(has_auth())`?

Reply:
It is decided to not allow `has_auth()` for consistencies.

https://github.com/AntelopeIO/spring/pull/1296/commits/675c6b25288ad1c1cbcb8345d77aa1cb6ab9cc07

Thanks.